src/server/http_server.rs

```diff
@@ -262,13 +272,17 @@ fn fail(
 
 /// A very literal, and rather unforgiving, implementation of RFC2616 (HTTP/1.1), returning the URL
 /// of GET requests: returns `Err` for anything else.
-fn parse_get(stream: &mut TcpStream) -> Result<Url, Box<dyn Error>> {
+fn parse_get<T: Read + Write>(
+    stream: &mut T,
+    http_port: u16,
```

Comment:
I suggest that rather than passing the ports (which does work, but which took me a while to reason about) we pass a `is_https: bool` flag which the (nicely separate) HTTP and HTTPS threads can easily set to true/false. That should (if I've got my thinking cap on correctly) simplify the check a little bit below to something like:

```rust
Some(h) => Url::parse(&format!("{}://{h:}{path:}", if is_https { "https" } else { "http" }))
  .map_err(|e| format!("Invalid request URI: {e:}").into()),
```